feature-libs/order/components/order-details/order-detail-items/consignment-tracking/consignment-tracking.component.ts

```diff
@@ -45,19 +62,25 @@ export class ConsignmentTrackingComponent implements OnInit, OnDestroy {
         consignment.code
       );
     }
-    let modalInstance: any;
-    this.modalRef = this.modalService.open(TrackingEventsComponent, {
-      centered: true,
-      size: 'lg',
-    });
-
-    modalInstance = this.modalRef.componentInstance;
-    modalInstance.tracking$ = this.consignmentTracking$;
-    modalInstance.shipDate = consignment.statusDate;
-    modalInstance.consignmentCode = consignment.code;
+    let modalInstanceData = {
+      tracking$: this.consignmentTracking$,
+      shipDate: consignment.statusDate,
+    };
+
+    const dialog = this.launchDialogService.openDialog(
+      LAUNCH_CALLER.CONSIGNMENT_TRACKING,
+      this.element,
+      this.vcr,
+      modalInstanceData
+    );
+
+    if (dialog) {
+      this.subscription.add(dialog.pipe(take(1)).subscribe());
```

Comment:
`query:` Do we need to use `subscription` in this case? I believe `take(1)` is handling unsubscribe under the hood. Consider to remove `Subscription` and `OnDestroy` interface.